File: core/src/rustyice/screens/LoadingScreen.java
package rustyice.screens;

import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.scenes.scene2d.Stage;
import com.badlogic.gdx.scenes.scene2d.ui.Container;
import com.kotcrab.vis.ui.widget.VisProgressBar;
import rustyice.core.Core;
import rustyice.screens.effects.GuiEffects;

public class LoadingScreen extends Screen{
    private Container<VisProgressBar> root;
    private VisProgressBar progressBar;
    
    @Override
    public void load() {
        super.load();
        
        progressBar = new VisProgressBar(0, 1, 0.01f, false);
        root = new Container<>(progressBar);
        root.setFillParent(true);
    }

    @Override
    public void show(Stage stage) {
        getStage().addActor(root);
    }

    @Override
    public void hide(Stage stage) {
        GuiEffects.fadeOut(root, 0.5f, () ->
            stage.getActors().removeValue(root, true)
        ).start(getTweenManager());
    }

    @Override
    public void dispose() {
    }

    @Override
    public void resize(int width, int height) {
    }

    @Override
    public void render(SpriteBatch batch, float delta) {
        if(Core.resources.update()){
            Core.resources.loadAll();
            getManager().showScreen("main_menu");
        }
        
        progressBar.setValue(Core.resources.getProgress());
    }

}

File: core/src/rustyice/screens/LoadingScreen.kt
package rustyice.screens

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.scenes.scene2d.ui.Container
import com.kotcrab.vis.ui.widget.VisProgressBar
import rustyengine.RustyEngine
import rustyengine.resources.Resources

class LoadingScreen: Screen(){
    private val root: Container<VisProgressBar>
    private val progressBar: VisProgressBar
    
    init {
        progressBar = VisProgressBar(0f, 1f, 0.01f, false)
        root = Container(progressBar)
        root.setFillParent(true)
    }

    override fun show() {
        stage.addActor(root)
    }

    override fun hide() {
        stage.actors.removeValue(root, true)
    }

    override fun dispose() {}

    override fun resize(width: Int, height: Int) {}

    override fun render(batch: Batch, delta: Float) {
        if(RustyEngine.resorces.update()){
            RustyEngine.resorces.loadAll()
            screenManager.showScreen("main_menu")
        }
        
        progressBar.value = RustyEngine.resorces.progress
    }
}